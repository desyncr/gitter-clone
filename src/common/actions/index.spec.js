import shortid      from 'shortid';
import * as actions from './index';

describe('actions', () => {
  it('addMessage should create ADD_MESSAGE action', () => {
    const message = {
      username: 'Bob',
      text: 'Use Redux'
    };
    expect(actions.addMessage(message))
      .toEqual({
        type: 'server.ADD_MESSAGE',
        id: 0,
        username: 'Bob',
        text: 'Use Redux'
      });
  });

  it('fetchMessages should create [CALL_API] action', () => {
    const convo = 'chat';
    expect(actions.fetchMessages(convo))
    .toMatchSnapshot();
  });

  it('addConvo should create ADD_CONVO action', () => {
    const convo = {name: 'chat'};
    expect(actions.addConvo(convo))
      .toMatchSnapshot();
  });

  it('fetchConvos should create [CALL_API] action', () => {
    expect(actions.fetchConvos())
      .toMatchSnapshot();
  });

  describe('signIn action creator', () => {
    it('should create SIGN_IN action', () => {
      const getState = () => ('Bob');
      const dispatch = jest.fn();
      const name = 'Dan';
      const id = shortid.generate();
      actions.signIn(name, id)(dispatch, getState);
      expect(dispatch)
        .toHaveBeenCalledWith({
          type: 'SIGN_IN',
          username: name,
          id
        });
    });

    it('should create @@router/CALL_HISTORY_METHOD action', () => {
      const getState = () => ('Bob');
      const dispatch = jest.fn();
      actions.signIn()(dispatch, getState);
      expect(dispatch)
        .toHaveBeenCalledWith({
          payload: {
            args: ['/chat'],
            method: 'push'
          },
          type: '@@router/CALL_HISTORY_METHOD'
        });
    });
  });

  it('openCreateRoomModal should create OPEN_CREATE_ROOM_MODAL action', () => {
    expect(actions.openCreateRoomModal())
      .toEqual({
        type: 'OPEN_CREATE_ROOM_MODAL',
      });
  });

  it('closeModal should create CLOSE_MODAL action', () => {
    expect(actions.closeModal())
      .toEqual({
        type: 'CLOSE_MODAL',
      });
  });
});
