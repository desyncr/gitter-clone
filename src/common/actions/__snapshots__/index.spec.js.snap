// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions addConvo should create ADD_CONVO action 1`] = `
Object {
  "name": "chat",
  "type": "server.ADD_CONVO",
}
`;

exports[`actions fetchConvos should create [CALL_API] action 1`] = `
Object {
  Symbol(Call API): Object {
    "endpoint": "null/api/convos",
    "method": "GET",
    "types": Array [
      "CONVOS_REQUEST",
      Object {
        "payload": [Function],
        "type": "CONVOS_SUCCESS",
      },
      "CONVOS_FAILURE",
    ],
  },
}
`;

exports[`actions fetchMessages should create [CALL_API] action 1`] = `
Object {
  Symbol(Call API): Object {
    "endpoint": "null/api/messages/chat",
    "method": "GET",
    "types": Array [
      "MESSAGES_REQUEST",
      Object {
        "payload": [Function],
        "type": "MESSAGES_SUCCESS",
      },
      "MESSAGES_FAILURE",
    ],
  },
}
`;
